### PR TITLE
Un-deprecate Some Public Example Images

### PR DESCRIPTION
**Pull request by jatcod3r**
Continue support of example images used by our example templates repository, [coder/templates](https://github.com/coder/templates), as part of our effort to centralize Coder examples.

- Un-deprecates:
  - deprecated/node
  - deprecated/java
  - deprecated/golang

- Adds:
  - images/rstudio-server

### CODE REVIEW

**Review by ericpaulsen — APPROVED** [review]

File: images/golang/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Enterprise Golang
+# Example Golang
```
Comment (matifali) [inline images/golang/README.md:1]:
```suggestion
# Golang
```

File: images/java/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Enterprise Java
+# Example Java
```
Comment (matifali) [inline images/java/README.md:1]:
```suggestion
# Java
```

File: images/rstudio-server/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Example RStudio Server
```
Comment (matifali) [inline images/rstudio-server/README.md:1]:
```suggestion
# RStudio Server
```

File: images/node/README.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Enterprise Node
+# Example Node
```
Comment (matifali) [inline images/node/README.md:1]:
```suggestion
# Node
```

**Review by matifali — APPROVED** [review]
A few suggestions but looks good to me.

File: images/golang/README.md (hunk 1 above)
Comment (jatcod3r) [reply images/golang/README.md]: Committed!

File: images/rstudio-server/README.md (hunk 3 above)
Comment (jatcod3r) [reply images/rstudio-server/README.md]: Removed the RStudio image. Moved IDE to a sub-module

File: images/node/README.md (hunk 4 above)
Comment (jatcod3r) [reply images/node/README.md]: Committed!

**Comment by matifali** [comment]
@jatcod3r, could you also make sure the deprecated images are being pushed to dockerhub?
I don't see them updated here: https://hub.docker.com/u/codercom

**Pull request merged**